### PR TITLE
feat(react): upgrade ExpandableTile from class to function

### PR DESCRIPTION
**Pull request by abbeyhrt**
Partially addresses #9712 

Updates ExpandableTile to functional component and updates the tests since they relied on the component being a class. 

#### Changelog

**Changed**

- tests
- ExpandableTile class -> function

#### Testing / Reviewing

Check that ExpandableTile works the same as it currently does

### CODE REVIEW

File: packages/react/src/components/Tile/next/Tile.js [hunk 1]
```diff
@@ -398,307 +392,267 @@ SelectableTile.propTypes = {
...
+  /**
+   * An ID that can be provided to aria-labelledby
+   */
+  id: PropTypes.string,
+
+  /**
+   * `true` to use the light version. For use on $ui-01 backgrounds only.
+   * Don't use this to make tile background color same as container background color.
+   */
+  light: PropTypes.bool,
```
Comment (sstrubberg) [inline packages/react/src/components/Tile/next/Tile.js]: the light prop needs to be deprecated

File: packages/react/src/components/Tile/next/Tile.js [hunk 2]
```diff
@@ -398,307 +392,267 @@ SelectableTile.propTypes = {
...
+TileAboveTheFoldContent.displayName = 'TileAboveTheFoldContent';
 
-    return (
-      <span ref={ref} className={`${prefix}--tile-content__below-the-fold`}>
-        {children}
-      </span>
-    );
-  }
-);
+export function TileBelowTheFoldContent({ children }) {
```
Comment (sstrubberg) [inline packages/react/src/components/Tile/next/Tile.js]: Why rip out the `React.forwardRef()`?

Comment (abbeyhrt) [reply packages/react/src/components/Tile/next/Tile.js]: Not intentional! I accidentally was working on a branch without your changes merged in and didn't catch this when I merged in main, good catch! Thanks!

**Review by sstrubberg — CHANGES_REQUESTED** [review]
Looks great! How did you know to dig up that useIsomorphicEffect hook?!

I think all you need is to replace your function with the `React.fowardRef()`.

**Review by tay1orjones — APPROVED** [review]

**Review by sstrubberg — APPROVED** [review]

**Pull request merged**